src/test/java/io/wgj/ctci/ArrayListTest.java

```diff
@@ -0,0 +1,136 @@
+package io.wgj.ctci;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.*;
+
+public class ArrayListTest {
+  @Rule
+  public final ExpectedException exception = ExpectedException.none();
+
+  @Test
+  public void arrayList_new() {
+    ArrayList<Integer> al = new ArrayList<>();
+  }
+
+  @Test
+  public void arrayList_add() {
+    ArrayList<Integer> al = new ArrayList<>();
+    al.add(1);
+  }
+
+  @Test
+  public void arrayList_size() {
+    ArrayList<Integer> al = new ArrayList<>();
+    assertEquals(0, al.size());
+    al.add(1);
+    assertEquals(1, al.size());
+  }
```

Comment:
This is a perfect example of what I'm talking about.  If you have two tests that that are exactly the same except one has extra statements, it almost always the case that one of the them is unnecessary.  Coverage can actually be just duplication if it doesn't help you find problems.

Reply:
Gotcha. I'm starting to think of tests more for their coverage than discrete little experiments.